kubetest/kubeadmdind/kubeadm_dind.go

```diff
@@ -0,0 +1,567 @@
+	nodeLogDir = "/var/log"
+
+	// Relative path to Kubernetes source tree
+	kubeOrg  = "k8s.io"
+	kubeRepo = "kubernetes"
+
+	// Kubeadm-DinD-Cluster (kdc) repo and main script
+	kdcOrg    = "github.com/kubernetes-sigs"
+	kdcRepo   = "kubeadm-dind-cluster"
+	kdcScript = "fixed/dind-cluster-stable.sh"
+
+	// Number of worker nodes to create for testing
+	numWorkerNodes = "2"
+)
+
+// Deployer is used to implement a kubetest deployer interface
+type Deployer struct {
+	ipv6Only  bool
+	hostCmder execCmder
+	control   *process.Control
+}
+
+// NewDeployer returns a new Kubeadm-DinD Deployer
+func NewDeployer(ipMode string, control *process.Control) (*Deployer, error) {
+	if ipMode == "dual-stack" {
+		return nil, fmt.Errorf("configured --ip-mode=%s is not supported for --deployment=gce-dind", ipMode)
+	}
+	ipv6Only := ipMode == "ipv6"
+
+	d := &Deployer{
+		ipv6Only:  ipv6Only,
+		hostCmder: new(hostCmder),
+		control:   control,
+	}
+
+	return d, nil
+}
+
+// execCmd executes a command on the host container.
+func (d *Deployer) execCmd(cmd string) *exec.Cmd {
+	return d.hostCmder.execCmd(cmd)
+}
+
+// runWithEnv runs a command on the host container with environment
+// variable settings, and prints any errors.
+func (d *Deployer) runWithEnv(cmd string, env []string) error {
+	execCmd := d.execCmd(cmd)
+	execCmd.Env = env
+	err := d.control.FinishRunning(execCmd)
+	if err != nil {
+		fmt.Printf("Error: '%v'", err)
+	}
+	return err
+}
+
+// run runs a command on the host container, and prints any errors.
+func (d *Deployer) run(cmd string) error {
+	return d.runWithEnv(cmd, []string{})
+}
+
+// getOutput runs a command on the host container, prints any errors,
+// and returns command output.
+func (d *Deployer) getOutput(cmd string) ([]byte, error) {
+	execCmd := d.execCmd(cmd)
+	o, err := d.control.Output(execCmd)
+	if err != nil {
+		log.Printf("Error: '%v'", err)
+		return nil, err
+	}
+	return o, nil
+}
+
+// outputWithStderr runs a command on the host container and returns
+// combined stdout and stderr.
+func (d *Deployer) outputWithStderr(cmd *exec.Cmd) ([]byte, error) {
+	var stdOutErr bytes.Buffer
+	cmd.Stdout = &stdOutErr
+	cmd.Stderr = &stdOutErr
+	err := d.control.FinishRunning(cmd)
+	return stdOutErr.Bytes(), err
+}
+
+// Up brings up a multinode, containerized Kubernetes cluster inside a
+// Prow DinD container.
+func (d *Deployer) Up() error {
+
+	d.setEnv()
+
+	// Scripts must be run from Kubernetes source directory
+	kubeDir, err := findPath(kubeOrg, kubeRepo, "")
+	if err == nil {
+		err = os.Chdir(kubeDir)
+	}
+	if err != nil {
+		return err
+	}
+
+	// Bring up a cluster inside the host Prow container
+	script, err := findPath(kdcOrg, kdcRepo, kdcScript)
+	if err != nil {
+		return err
+	}
+	cmd := script + " up"
+	env := []string{"DIND_STORAGE_DRIVER=vfs"}
+	if err := d.runWithEnv(cmd, env); err != nil {
+		return err
+	}
+
+	// Build e2e.test and ginkgo binaries
```

Comment:
Probably move this to TestSetup()?

Reply:
Makes sense. I'll move these builds of e2e/ginkgo to TestSetup, and test this locally.